### PR TITLE
UI Update + Additional Features

### PR DESCRIPTION
**Pull request by JonathanLangton1**
### Features:
- Order summary data refresh every 4 seconds (including fee)
- Goby add asset to wallet (Hoogii doesn't support this yet)
- Goby includes Tibet suggested fee automatically within wallet integration (Hoogii to come soon when they support it)
- Data no longer continues to update after submitting the offer
- Added theme selector (system, force light, force dark) hidden within arrow toggle in footer
- If Goby is connected, it will now display a truncated value of your address in the connected wallet button
- Made things prettier ✨

### CODE REVIEW

File: src/components/GenerateOffer.tsx [hunk 1]
```diff
@@ -242,6 +263,12 @@ const GenerateOffer: React.FC<GenerateOfferProps> = ({ data, setOrderRefreshActi
         });
     };
 
+    const addAssetToWallet = async (assetId: string, symbol: string, logo: string) => {
+        if (!activeWallet) return console.log('Connect to a wallet before trying to add an asset')
```
Comment (Yakuhito) [inline src/components/GenerateOffer.tsx:268]: shouldn't this be a toast message?

File: src/components/GenerateOfferButton.tsx [hunk 2]
```diff
@@ -8,10 +8,10 @@ const GenerateOfferButton: React.FC<GenerateOfferButtonProps> = ({ disabled, onP
   return (
     <button
       onClick={onPressed}
-      className={`${disabled ? 'bg-brandDark/10 text-brandDark/20 dark:text-brandLight/30 cursor-not-allowed' : (isBuySelected ? 'bg-green-800' : 'bg-red-700')} text-brandLight px-4 py-2 rounded-lg w-full mt-8 font-medium`}
+      className={`${disabled ? 'bg-brandDark/10 text-brandDark/20 dark:text-brandLight/30 cursor-not-allowed' : (isBuySelected ? 'bg-green-800 hover:opacity-95' : 'bg-red-700 hover:opacity-95')} text-brandLight p-4 rounded-xl w-full mt-8 font-medium`}
       disabled={disabled}
     >
-      Generate Offer
+      Swap
```
Comment (Yakuhito) [inline src/components/GenerateOfferButton.tsx]: maybe have the text as an argument? wouldn't say 'Swap' for add/remove liquidity

File: src/utils/walletIntegration/wallets/hoogiiWallet.tsx [hunk 3]
```diff
@@ -110,6 +110,14 @@ class hoogiiWallet implements WalletIntegrationInterface {
     // Hoogii wallet balance retrieval logic
     console.log('Getting Hoogii Wallet Balance')
   }
+
+  getAddress() {
+    
+  }
+
+  async addAsset(assetId: string, symbol: string, logo: string): Promise<void> {
+    console.log('Currently only Goby has support for adding an asset programatically')
```
Comment (Yakuhito) [inline src/utils/walletIntegration/wallets/hoogiiWallet.tsx:119]: toast?

File: src/utils/walletIntegration/wallets/walletConnect.ts [hunk 4]
```diff
@@ -177,6 +177,14 @@ class WalletConnectIntegration implements WalletIntegrationInterface {
     // WalletConnect balance retrieval logic
   }
 
+  async addAsset(assetId: string, symbol: string, logo: string): Promise<void> {
+    console.log('Currently only Goby has support for adding an asset programatically')
```
Comment (Yakuhito) [inline src/utils/walletIntegration/wallets/walletConnect.ts:181]: toast? also, I'm pretty sure Chia has it, too

**Review by Yakuhito — CHANGES_REQUESTED** [review]

**Review by Yakuhito — APPROVED** [review]

File: src/utils/walletIntegration/wallets/walletConnect.ts (hunk 4 above)
Comment (JonathanLangton1) [reply src/utils/walletIntegration/wallets/walletConnect.ts:181]: I pasted the code from Hoogii. Will be built out when I work on WalletConnect again

File: src/components/GenerateOfferButton.tsx (hunk 2 above)
Comment (Yakuhito) [reply src/components/GenerateOfferButton.tsx]: Update: did it myself

File: src/utils/walletIntegration/wallets/hoogiiWallet.tsx (hunk 3 above)
Comment (JonathanLangton1) [reply src/utils/walletIntegration/wallets/hoogiiWallet.tsx:119]: This function will never be hit as "add asset to wallet" button only shows for supported wallets. Added it here for consistency. Now that I think of it, I may as well just convert this to a toast if it somehow gets hit 👍

File: src/components/GenerateOffer.tsx (hunk 1 above)
Comment (JonathanLangton1) [reply src/components/GenerateOffer.tsx:268]: Yep thanks

**Pull request merged**